### PR TITLE
docs: fix Home Assistant MQTT sensor JSON templates warnings

### LINKED ISSUE #4247: home-assistant template errors
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Is starting to urk me that home-assistant is generating non-stop errors on my system. This aren't actually "errors" in the sense of something going wrong, just annoying to look at in the logs.

It would look appear the biggest culprit is the teslamate recommended config from https://docs.teslamate.org/docs/integrations/home_assistant/

Unfortunately, my home-assistant knowledge in fixing these isn't particularly great.

(on a semi-related note, would be really good id teslamate could send out discovery messages so that this can be auto-configured - see https://www.home-assistant.io/integrations/mqtt#mqtt-discovery - assuming this is possible)

### Expected Behavior

No errors

### Steps To Reproduce

Run home-assistant after adding settings from https://docs.teslamate.org/docs/integrations/home_assistant/

### Relevant log output

```shell
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.components.mqtt.mixins] TypeError: Object of type LoggingUndefined is not JSON serializable rendering template for entity 'device_tracker.tesla_active_route_location', template: '{{ value_json.location | tojson }}' and payload: {"error":"No active route available"}
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'destination' when rendering '{{ value_json.destination }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'energy_at_arrival' when rendering '{{ value_json.energy_at_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'miles_to_arrival' when rendering '{{ value_json.miles_to_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'minutes_to_arrival' when rendering '{{ value_json.minutes_to_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'traffic_minutes_delay' when rendering '{{ value_json.traffic_minutes_delay }}'
Oct 07 15:10:45 iot2 homeassistant[107792]: 2024-10-07 15:10:45.723 ERROR (Thread-9) [pychromecast.socket_client] [Dining room TV(192.168.5.192):8009] Failed to connect to service MDNSServiceInfo(name='Chromecast-c6be7577e2d8f2215bf00c76a003635b._googlecast._tcp.local.'), retrying in 5.0s
Oct 07 15:10:46 iot2 homeassistant[107792]: 2024-10-07 15:10:46.176 ERROR (MainThread) [homeassistant.helpers.event] Error while processing template: Template<template=({{ (states('sensor.tesla_active_route_distance_to_arrival_mi') | float * 1.609344) | round(2) }}) renders=2>
```

```
Oct 07 15:10:46 iot2 homeassistant[107792]: homeassistant.exceptions.TemplateError: ValueError: Template error: float got invalid input 'unavailable' when rendering template '{{ (states('sensor.tesla_active_route_distance_to_arrival_mi') | float * 1.609344) | round(2) }}' but no default was specified
Oct 07 15:10:46 iot2 homeassistant[107792]: 2024-10-07 15:10:46.179 ERROR (MainThread) [homeassistant.components.template.template_entity] TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{{ (states('sensor.tesla_active_route_distance_to_arrival_mi') | float * 1.609344) | round(2) }}' but no default was specified') while processing template 'Template<template=({{ (states('sensor.tesla_active_route_distance_to_arrival_mi') | float * 1.609344) | round(2) }}) renders=4>' for attribute '_attr_native_value' in entity 'sensor.tesla_active_route_distance_to_arrival_km'
```
```


### Screenshots

_No response_

### Additional data

_No response_

### Type of installation

Manual

### Version

latest

### PR DESCRIPTION
**Pull request by longzheng**
Fixes #4247

More gracefully handles `teslamate/cars/1/active_route` topic JSON when the car is not routing, the `json_attributes_template` would cause a Home Assistant warning since we're trying to access properties on the JSON that aren't there.

MQTT topic `active_route`
```
{"error":"No active route available"}
```

Home Assistant logs
```
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.components.mqtt.mixins] TypeError: Object of type LoggingUndefined is not JSON serializable rendering template for entity 'device_tracker.tesla_active_route_location', template: '{{ value_json.location | tojson }}' and payload: {"error":"No active route available"}
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'destination' when rendering '{{ value_json.destination }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.127 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'energy_at_arrival' when rendering '{{ value_json.energy_at_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'miles_to_arrival' when rendering '{{ value_json.miles_to_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'minutes_to_arrival' when rendering '{{ value_json.minutes_to_arrival }}'
Oct 07 15:10:43 iot2 homeassistant[107792]: 2024-10-07 15:10:43.128 WARNING (MainThread) [homeassistant.helpers.template] Template variable warning: 'dict object' has no attribute 'traffic_minutes_delay' when rendering '{{ value_json.traffic_minutes_delay }}'
```

Also fix other template sensor errors mentioned in https://github.com/teslamate-org/teslamate/issues/4247#issuecomment-2396554423 which are unrelated to the active route error but a similar issue where the template reference other MQTT sensor numeric values which may be invalid/unavailable, which causes a warning when trying to use math expressions on.

### CODE REVIEW

**Comment by swiffer** [comment]
tested successfully here (updated mqtt_sensors.yaml, restarted home_assistant, no errors in logs).

File: website/docs/integrations/home_assistant.md [hunk 1]
```diff
@@ -648,84 +668,117 @@ Don't forget to replace `<teslamate url>` and `<your tesla model>` with correct
...
 
     tesla_elevation_ft:
       friendly_name: Elevation (ft)
       unit_of_measurement: ft
       icon_template: mdi:image-filter-hdr
+      availability_template: "{{ has_value('sensor.tesla_elevation') }}"
       value_template: >
-       {{ (states('sensor.tesla_elevation') | float * 3.2808 ) | round(2) }}
+        {% if has_value('sensor.tesla_elevation') %}
+          {{ (states('sensor.tesla_elevation') | float / 3.2808) | round(2) }}
```
Comment (brianmay) [inline website/docs/integrations/home_assistant.md:730]: Did you mean to change the * operation to divide here?

File: website/docs/integrations/home_assistant.md [hunk 2]
```diff
@@ -648,84 +668,117 @@ Don't forget to replace `<teslamate url>` and `<your tesla model>` with correct
...
 
     tesla_tpms_pressure_fl_psi:
       friendly_name: Front Left Tire Pressure (psi)
       unit_of_measurement: psi
       icon_template: mdi:car-tire-alert
+      availability_template: "{{ has_value('sensor.tesla_tpms_fl') }}"
       value_template: >
-       {{ (states('sensor.tesla_tpms_pressure_fl') | float * 14.50377) | round(2) }}
+        {% if has_value('sensor.tesla_tpms_fl') %}
+          {{ (states('sensor.tesla_tpms_fl') | float / 14.50377) | round(2) }}
```
Comment (brianmay) [inline website/docs/integrations/home_assistant.md:740]: Same thing for all the tire pressures.

**Review by brianmay — CHANGES_REQUESTED** [review]
Looks good, but a bit puzzled about the changes from multiple to divide in some places.

File: website/docs/integrations/home_assistant.md (hunk 1 above)
Comment (longzheng) [reply website/docs/integrations/home_assistant.md:730]: Sorry mistake from copy paste.

**Review by brianmay — APPROVED** [review]

**Comment by longzheng** [comment]
@JakobLichterfeld not sure why the CI failed, can you retry please

**Comment by JakobLichterfeld** [comment]
> @JakobLichterfeld not sure why the CI failed, can you retry please

Yeah, GitHub Runners have been slow yesterday and caused lots of failures, not related to your PR. Today it runs fine.

**Pull request merged**